### ISSUE TITLE
Removing Last Item from ObeservableCollection for ListView.ItemsSource throws exception on Windows 8.1 (Desktop)

### ISSUE THREAD

**Issue by mattclarkie** [xamarin/Xamarin.Forms#1322]
_The issue https://bugzilla.xamarin.com/show_bug.cgi?id=59650 is still present on Windows RT (8.1 Desktop). As this is merely a re-open of an already known issue please refer to the original bugzilla for exact details._

Removing the last item from an Observable Collection bound to a ListView will throw an ArgumentOutOfRangeException when running on Windows 8.1, present in version 2.5.0.91635 of Xamarin.Forms.

```
   at System.Linq.Enumerable.ElementAt[TSource](IEnumerable`1 source, Int32 index)
   at Xamarin.Forms.ListView.NotifyRowTapped(Int32 groupIndex, Int32 inGroupIndex, Cell cell)
   at Xamarin.Forms.ListView.NotifyRowTapped(Int32 index, Cell cell)
   at Xamarin.Forms.Platform.WinRT.ListViewRenderer.OnListItemClicked(Int32 index)
   at Xamarin.Forms.Platform.WinRT.ListViewRenderer.OnControlSelectionChanged(Object sender, SelectionChangedEventArgs e)
   at System.Collections.Specialized.NotifyCollectionChangedEventHandler.Invoke(Object sender, NotifyCollectionChangedEventArgs e)
   at System.Collections.ObjectModel.ObservableCollection`1.OnCollectionChanged(NotifyCollectionChangedEventArgs e)
   at System.Collections.ObjectModel.ObservableCollection`1.RemoveItem(Int32 index)
   at System.Collections.ObjectModel.Collection`1.Remove(T item)
   at RFA.Views.ClientViewModel.<>c__DisplayClass91_0.<RemoveNote>b__0()
```

**Comment by pauldipietro** [comment]
Windows 8.1 is no longer supported, so this issue will be closed. If this affects UWP as well, please reopen it with a reproduction. Also, in the future, please aim to use the provided issue reporting template. Thanks!

**Comment by mattclarkie** [comment]
Windows 8.1 is still supported on the Xamarin 2 branch. Our application is unusable due to this.

**Comment by davidortinau** [comment]
@mattclarkie from the bugzilla report it appears that issue was resolved and confirmed in a nightly build. Did you check that? 

If that's the case and this is already fixed, then we could make sure that it is included in the next 2.5.0 service release.

**Comment by mattclarkie** [comment]
@davidortinau the person who verified it as "resolved" "forgot" that Windows 8.1 exists, the renderer affected is full of #if !WINDOWS_UWP and it's clearly one of those blocks that's causing the error, the UWP path presumably works perfectly fine.

Please state either
* WinRT is dead, broken, all bets are off
* Actually provide some reasonable level of support, and importantly  test for regressions

We are in the process of moving to UWP, in fact we've been working on the UWP port on and off for months and find xxx doesn't work in UWP or needs some extra workaround the other platforms don't. Our project started off using Xamarin.Forms 1.3.1.6296, we've put up with a lot over the years as Xamarin matured because it was fundamentally a good technology, but this experience, and your support processes make me question whether becoming so locked into your platform is a terrible mistake to make.

**Comment by davidortinau** [comment]
@mattclarkie I believe this has already been stated. The [end of life for support of Windows Phone](https://support.microsoft.com/en-us/help/4001737/products-reaching-end-of-support-for-2017?ranMID=24542&ranEAID=nOD%2FrLJHOac&ranSiteID=nOD_rLJHOac-E7AOXKep5sgi0l8P5nA3vg&tduid=(9102533867c4bcd2799744e6bb6e5d3c)(256380)(2459594)(nOD_rLJHOac-E7AOXKep5sgi0l8P5nA3vg)()) has been public for a long time. I'm sorry if our product communication on the impending removal of these platforms was not made more clear. For months we have been marking all Windows Phone 8/8.1 issues as "won't fix".

That said, we absolutely desire to support you and help you through this transition. 

If the nightly build is confirmed to address your issue, I'll work with engineering to see that fix included in the next service release.

Please email me your UWP frustrations and identify the related Bugzilla or Issues so I can help prioritize those issues and/or provide guidance. david.ortinau@microsoft.com

**Comment by mattclarkie** [comment]
@davidortinau 
Firstly, this is not Windows Phone, this is Windows Desktop 8.1. They are not the same thing! Windows 8.1 is still in mainstream support until January 2018 so Xamarin's decision to pull the plug on it alongside Windows Phone is barely justifiable.

Secondly, you only introduced the issue a few months ago (October is my best guess from looking at your source), and now are washing your hands of it. If Windows 8.1 is EoL then please stop breaking the existing functionality, you could have left the v2.0 branch well alone instead of breaking it for those of us that are still using Windows RT. Deprecated doesn't mean actively break the existing functionality. We've upgraded to latest stable to get bug fixes required for UWP and the result is the Windows 8.1 application is so broken it can't be released thanks to "latest [un]stable".

**Comment by davidortinau** [comment]
@mattclarkie apologies, you're right. Thanks for correcting me. They are not the same thing.

We have not removed WinRT from the [2.5.0 branch](https://github.com/xamarin/Xamarin.Forms/tree/15-5).

Our next major release will not include WinRT and Windows Phone 8/8.1 backends and should be expected to occur sometime after the [January 2018 mark](https://support.microsoft.com/en-us/help/18581/lifecycle-faq-windows-products).  

For any future, long term work to these platforms, the best option is to fork and work from the 2.5.0 branch linked above.

In terms of how we can help in the immediate for this issue that's blocking you, have you been able to test a [nightly release](https://github.com/xamarin/Xamarin.Forms/wiki/Nightly-Builds)? Another option is for you to PR a fix and we'll get it into a 2.5.0 service release.

In terms of your Windows 8.1. and UWP situation, it is possible to configure your solutions to run different versions of Xamarin.Forms. It doesn't sound fun, but if it solve you situation until you can move completely to UWP, then perhaps it's worth it.
